### PR TITLE
Adding e2e tests for pages which use middleware to redirect

### PR DESCRIPTION
**Pull request by codyrancher**
### Summary
Also fixed a couple routes, and removed some test setup code that was commented on in another review.

### Technical notes summary
I wanted to put these tests in place before changing or removing the middleware to make sure we keep the experience consistent.

### Areas or cases that should be tested
These are the tests.

### Areas which could experience regressions
None

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/config/router.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ export const routerOptions = {
           name:      'account-create-key'
         },
         {
-          path:      'auth',
```
Comment (codyrancher) [inline shell/config/router.js:92]: This configuration wasn't actually being used correctly. It was falling back on `/c/:cluster/:product`

auth

File: shell/config/router.js [hunk 2]
```diff
@@ -99,7 +99,7 @@ export const routerOptions = {
           name:      'c-cluster-ecm'
         },
         {
-          path:      'settings',
```
Comment (codyrancher) [inline shell/config/router.js:102]: This configuration wasn't actually being used correctly. It was falling back on /c/:cluster/:product

settings

File: jest.setup.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ global.TextEncoder = TextEncoder;
 global.TextDecoder = TextDecoder;
 
 Vue.config.productionTip = false;
-Vue.use(i18n, { store: { dispatch() {} } });
```
Comment (codyrancher) [inline jest.setup.js:17]: Made this change in regards to https://github.com/rancher/dashboard/pull/10923#discussion_r1588528484

**Review by cnotv — COMMENTED** [review]
For me, the test coverage seems fine, but I do not grasp why it should redirect and if this is common in all these pages, would it not be better a unit test at this point?
It obviously does not exclude having both or any issue to have just E2E, so it's just a simple curiosity.

**Comment by codyrancher** [comment]
> For me, the test coverage seems fine, but I do not grasp why it should redirect and if this is common in all these pages, would it not be better a unit test at this point? It obviously does not exclude having both or any issue to have just E2E, so it's just a simple curiosity.

Most of these redirects don't have any logic so I was planning to move them as redirects defined in the router config. This would change the units all together so I'd have to rewrite all the unit tests and therefore I wouldn't be verifying that I didn't change behavior.

I could just make the changes and maybe write some router unit tests without this verification step. I'm okay with either. Let me know which you'd prefer.

**Comment by cnotv** [comment]
> Most of these redirects don't have any logic so I was planning to move them as redirects defined in the router config. This would change the units all together so I'd have to rewrite all the unit tests and therefore I wouldn't be verifying that I didn't change behavior.

I would expect just to change the spied function, although I may completely miss the actual case.
Could we at least specify when the redirect happens in the test? Because as it's written it simply redirects for no reason.
Is it me overlooking something or would you agree? If not agree, please leave it as it is :)
 
> I could just make the changes and maybe write some router unit tests without this verification step. I'm okay with either. Let me know which you'd prefer.

I tend to stick to an already done work, especially if is also fine.

**Review by cnotv — APPROVED** [review]
LGTM

**Pull request merged**